components/brave_adaptive_captcha/url_loader.cc

```diff
@@ -0,0 +1,147 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_adaptive_captcha/url_loader.h"
+
+#include <utility>
+
+#include "chrome/browser/browser_process.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+#include "services/network/public/cpp/resource_request.h"
+
+// Maximum size of the response in bytes.
+const int kMaxResponseSizeBytes = 1024 * 1024;
+
+namespace {
+
+std::string UrlMethodToRequestType(
+    brave_adaptive_captcha::UrlLoader::UrlMethod method) {
+  switch (method) {
+    case brave_adaptive_captcha::UrlLoader::UrlMethod::GET: {
+      return "GET";
+    }
+
+    case brave_adaptive_captcha::UrlLoader::UrlMethod::POST: {
+      return "POST";
+    }
+
+    case brave_adaptive_captcha::UrlLoader::UrlMethod::PUT: {
+      return "PUT";
+    }
+  }
```

Comment:
Nit:
```
default:
  NOTREACHED();
```
Also, you don't need the braces here.

Reply:
Thanks, fixed.

Reply:
A great article on why single line `if`/`case` statements are not safe https://medium.com/@jonathanabrams/single-line-if-statements-2565c62ff492